erpnext/regional/doctype/import_supplier_invoice/import_supplier_invoice.py

```diff
@@ -0,0 +1,396 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+from decimal import Decimal
+import json
+import re
+import traceback
+import zipfile
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.custom.doctype.custom_field.custom_field import create_custom_field
+from frappe.utils.data import format_datetime
+from bs4 import BeautifulSoup as bs
+from frappe.utils import cint, flt, today, nowdate, add_days, get_files_path
+import dateutil
+from frappe.utils.file_manager import save_file
+
+class ImportSupplierInvoice(Document):
+
+	def autoname(self):
+		if not self.name:
+			self.name = "Import Invoice on " + format_datetime(self.creation)
+
+	def import_xml_data(self):
```

Comment:
This method is too big, can we split the codes into multiple small functions?

Reply:
We are mainly parsing through file in this method so may not be possible I think but open to suggestions